icekit/bin/entrypoint.sh

```diff
@@ -31,15 +31,23 @@ if [[ -n "$DOCKER" ]]; then
     export -f pip
 
     # Use alternate installation (user scheme) for Python packages.
-    export PIP_SRC="$ICEKIT_PROJECT_DIR/var/venv/src"
-    export PYTHONUSERBASE="$ICEKIT_PROJECT_DIR/var/venv"
+    export PYTHONUSERBASE="$ICEKIT_PROJECT_DIR/var/docker-pyuserbase"
+    export PIP_SRC="$PYTHONUSERBASE/src"
+
+    # Add user scheme bin directory to PATH.
+    export PATH="$PYTHONUSERBASE/bin:$PATH"
 
     # For some reason pip allows us to install sdist packages, but not editable
     # packages, when this directory doesn't exist. So make sure it does exist.
     mkdir -p "$PYTHONUSERBASE/lib/python2.7/site-packages"
 else
     # Add ICEkit bin directory to PATH.
     export PATH="$ICEKIT_DIR/bin:$PATH"
+
+    # Prefer ICEKit virtualenv binaries to run things inside the venv
+    if [[ -n "$ICEKIT_VENV" ]]; then
+        export PATH="$ICEKIT_VENV/bin:$PATH"
+    fi
```

Comment:
We could only check for missing dependencies in the if/else block at the top of `entrypoint.sh` when running outside of Docker only. Inside Docker, we should be able to safely assume that all dependencies are available.

We could write our own Python implementation of `md5sum`, but we wouldn't need it when using Docker, and I don't have a problem requiring that everything needed by our scripts is manually installed when not using Docker. A bunch of other things much heavier than pv or md5sum are already required.

Running with Docker should be the ideal, and running without Docker should be possible if your system/env meets the requirements.

I'd consider ditching `pv` entirely instead of coding in conditional support for it, if you think it's not a useful thing. It's not providing any essential functionality. I just thought it was easier to not get lost in what is happening at startup to hide the raw SQL output, and convenient to see a ballpark progress bar showing how many MB has been piped so far (e.g. when restoring large production DB dumps for local testing).